src/services/fdp-storage.provider.ts

```diff
@@ -0,0 +1,85 @@
+import { FdpStorage } from '@fairdatasociety/fdp-storage'
+import { networks } from '../constants/networks'
+import { Network } from '../model/storage/network.model'
+import { Swarm } from '../model/storage/swarm.model'
+import { SwarmExtension } from '../swarm-api/swarm-extension'
+import { AsyncConfigService } from './async-config.service'
+import { Storage } from './storage/storage.service'
+
+export class FdpStorageProvider extends AsyncConfigService<FdpStorage> {
+  private storage: Storage = new Storage()
+  private network: Network
+  private swarm: Swarm
+
+  constructor() {
+    super()
+    super.initialize(async () => {
+      const [network, swarm] = await Promise.all([this.storage.getNetwork(), this.storage.getSwarm()])
+      this.network = network
+      this.swarm = swarm
+
+      this.storage.onNetworkChange((network: Network) =>
+        super.updateConfigAsync(this.createFdpStorage((this.network = network), this.swarm)),
+      )
+
+      this.storage.onSwarmChange((swarm: Swarm) =>
+        super.updateConfigAsync(this.createFdpStorage(this.network, (this.swarm = swarm))),
+      )
+
+      return this.createFdpStorage(network, swarm)
+    })
+  }
+
+  public getService(): Promise<FdpStorage> {
+    return new Promise((resolve) => {
+      setTimeout(() => {
+        resolve(super.getConfig())
+      })
+    })
+  }
+
+  private async createFdpStorage(network: Network, swarm: Swarm): Promise<FdpStorage> {
+    const { beeApiUrl, beeDebugApiUrl } = await this.getBeeAddresses(swarm)
```

Comment:
`beeApiUrl` and `beeDebugApiUrl` also can change during the time for which we need an event from the swarm-extension. create an issue for this problem please

Reply:
Yes, that's true. But I'm not sure if there is a mechanism to subscribe to events from another extension. I'll have to check that.